client/src/progress-reporter.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as vscode from 'vscode';
+
+const EMPTY_DISPOSABLE = vscode.Disposable.from();
+
+class ProgressReporter implements vscode.Progress<unknown> {
+  private disposable: vscode.Disposable = EMPTY_DISPOSABLE;
+
+  report(value: unknown) {
+    this.disposable.dispose();  // clear the last message
+    this.disposable = vscode.window.setStatusBarMessage(`$(loading~spin) Angular: ${value}`)
```

Comment:
Wow, they have their own formatter for the message? (!)

Reply:
They do! Here's the reference docs: https://code.visualstudio.com/api/references/icons-in-labels
I should add the link in the comment as well.